sdk/eventhub/azure-eventhub/azure/eventhub/aio/_buffered_producer/_buffered_producer_async.py

```diff
@@ -137,14 +136,14 @@ async def _flush(self, timeout_time=None, raise_error=True):
         # try flushing all the buffered batch within given time
         _LOGGER.info("Partition: %r started flushing.", self.partition_id)
         if self._cur_batch:  # if there is batch, enqueue it to the buffer first
-            self._buffered_queue.put(self._cur_batch)
+            await self._buffered_queue.put(self._cur_batch)
             self._cur_batch = EventDataBatch(self._max_message_size_on_link, amqp_transport=self._amqp_transport)
         while self._buffered_queue.qsize() > 0:
             remaining_time = timeout_time - time.time() if timeout_time else None
             if (remaining_time and remaining_time > 0) or remaining_time is None:
                 try:
-                    batch = self._buffered_queue.get(block=False)
-                except queue.Empty:
+                    batch = self._buffered_queue.get_nowait()
+                except asyncio.QueueEmpty:
                     break
```

Comment:
do we know why we were using sync queue?

Reply:
no idea why that was the case.